content/Voice Over - Fernanda Toledo.md

```diff
@@ -0,0 +1,63 @@
+
+### Get your hands dirty
+
+One fun little experiment we did at Rootstrap was developing a very silly app. The purpose of it was to create to-do lists, users could add items that consisted of just a checkbox and a text and they could mark them as either complete or incomplete by checking on the checkbox, classic to-do stuff. We handed a blindfold to our CTO and asked him to add one item and mark it as complete, it was painful to watch, he struggled for a while to figure out how to open the keyboard. Then, we improved the default implementation a little:
+
+- Gave our UI an intuitive hierarchy structure: Group UI elements that make sense together to provide a faster and better navigation
+
+- isAccessibleElement: Make non relevant UI elements not focusable by Voice Over, like shadows and images, anything that's decorative
+
+- accessibilityLabel: Provide a useful description to Voice Over to read when the item is focused
+
+- accessibilityHint: Provide additional context or actions for the selected element
+
+- accessibilityTraits: Tells Voice Over how the elements behave or should be treated, this can be button, isSelected, text, etc
+
+- UIAccessibilityLayoutChangedNotification: Notifies Voice Over about changes in layout, so that the last item added to the list gets focused
+
+
+These improvements that didn't take long to develop made a huge difference in UX, and we were able to prove it with our guinea pig who completed his mission successfully.
+
+![](https://github.com/rootstrap/blog/blob/voice-over/content/images/blindfold.jpg)
+
+### Debug it
+
+The Accessibility Inspector belongs to the deck of developer tools, and one of the most powerful features provided by this tool is the ability to simulate VoiceOver, this was introduced in [xCode11](https://developer.apple.com/videos/play/wwdc2019/257/#:~:text=Menu%20Close%20Menu-,Accessibility%20Inspector,Voice%20Over%20user%20would%20experience.). And it's fantastic. You can simulate Voice Over thanks to a feature that was added recently by tapping on the speaker button. The auto-navigate button will navigate through all accessible views and read them for you so that you can debug it. A basic example that will provide a poor experience for users is when there’s a custom button with an image and VoiceOver will read the image file name, because no accessibility label was provided.
+
+What's more, it does not only help you to detect issues with Voice Over, you can run an audit and it will throw any kind of accessibility issue. We invite you to try it on your own apps. It will detect more than you expect, potentially inaccessible text, contrast ratio, a button with a hit area too small, elements with no description, and unsupported dynamic fonts sizes.
```

Comment:
```suggestion
It does not only help you to detect issues with VoiceOver, but you can also run an audit and it will throw any kind of accessibility issue. We invite you to try it on your own apps. It will detect more than you expect, potentially inaccessible text, contrast ratio, a button with a hit area too small, elements with no description, and unsupported dynamic fonts sizes.
```